crates/model/src/order.rs

```diff
@@ -84,6 +85,73 @@ impl Order {
         token_list.contains(&self.creation.buy_token)
             || token_list.contains(&self.creation.sell_token)
     }
+
+    /// Returns the remaining amounts for the order.
+    ///
+    /// For fill-or-kill orders, this trivially returns full buy, sell and fee
+    /// amounts for the order (since the order can only either be not executed
+    /// at all or fully executed).
+    ///
+    /// For partially fillable orders, it considers the currently executed
+    /// amount and computes what is left in the order.
+    ///
+    /// Returns `None` on overflows.
+    pub fn remaining_amounts(&self) -> Result<Option<RemainingOrderAmounts>> {
+        let remaining_amounts = if self.creation.partially_fillable {
```

Comment:
I believe this would be a little better readable if there is an early return for "normal" orders instead of a huge if-else-block. That would eliminate 1 level of indentation.